### PR TITLE
release FFA-Net model

### PR DESCRIPTION
**Pull request by bitcjm**
Newly integrated functional modules related to the FFA-NET model and the repo has been modified as submitted

### CODE REVIEW

**Comment by bitcjm** [comment]
Newly integrated functional modules related to the FFA-NET model and the repo has been modified as submitted, fixed some small bugs.

File: README.md [hunk 1]
```diff
@@ -1,234 +1,287 @@
-[English](README_en.md) | 中文
```
Comment (huangjun12) [inline README.md:1]: 这个文档放这里吧 https://github.com/PaddlePaddle/PaddleVideo/tree/develop/docs/zh-CN/model_zoo/estimation

新建一个文件 ffa.md

File: README_en.md [hunk 2]
```diff
@@ -1,224 +1,285 @@
-[简体中文](README.md) | English
+##  [FFA-Net: Feature Fusion Attention Network for Single Image Dehazing](https://arxiv.org/abs/1911.07559) (AAAI 2020)
```
Comment (huangjun12) [inline README_en.md:1]: 这个文档放这里吧 https://github.com/PaddlePaddle/PaddleVideo/blob/develop/docs/en/model_zoo/estimation/

新建一个文件ffa.md

File: configs/FFA_cfg.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# PaddleVieo Example Configuration, please refer to "docs/en/config.md" for more information.
```
Comment (huangjun12) [inline configs/FFA_cfg.yaml:1]: 配置文件放这里 https://github.com/PaddlePaddle/PaddleVideo/tree/develop/configs/estimation/adds

File: configs/FFA_cfg.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+###commands###
+#python main.py --test -c configs/FFA_cfg.yaml -w data/FFA/ITS_3_19_article_pretrained.pdparams
```
Comment (huangjun12) [inline configs/FFA_cfg.yaml:3]: 这些命令注释都删了吧 
在文档里写明就行了

**Review by huangjun12 — COMMENTED** [review]
figs/image-20220409164858962.png 这张图像不用提到pr中，删了吧

File: paddlevideo/loader/dataset/ffa_dataset.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+#encoding=utf8
```
Comment (huangjun12) [inline paddlevideo/loader/dataset/ffa_dataset.py:1]: 这个删了

File: paddlevideo/loader/dataset/ffa_dataset.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    def __init__(self,
+                 pipeline,
+                 data_prefix="",
+                 file_path="",
+                 crop_size=240,
+                 test_mode=False,
+                 suffix='.png'):
+        super().__init__(file_path, pipeline)
+        #self.pipeline = pipeline
```
Comment (huangjun12) [inline paddlevideo/loader/dataset/ffa_dataset.py:35]: 删除无用注释

File: paddlevideo/modeling/backbones/ffa_net.py [hunk 7]
```diff
@@ -0,0 +1,202 @@
+#encoding=utf8
```
Comment (huangjun12) [inline paddlevideo/modeling/backbones/ffa_net.py:1]: 删除这个

File: paddlevideo/modeling/backbones/ffa_net.py [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+from paddle.nn import Conv2D, ReLU, AdaptiveAvgPool2D
+from paddle import nn
+from paddle.framework import ParamAttr
+from paddle.nn.initializer import Uniform
```
Comment (huangjun12) [inline paddlevideo/modeling/backbones/ffa_net.py:20]: 导入方式和后面的写法统一一下：

import paddle.nn as nn

File: paddlevideo/modeling/backbones/ffa_net.py [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+import math
+from ..registry import BACKBONES
+
+#定义参数初始化规则
+weight_ins = paddle.ParamAttr(initializer=paddle.nn.initializer.Normal(
```
Comment (huangjun12) [inline paddlevideo/modeling/backbones/ffa_net.py:25]: paddle.nn.initializer.Normal --> nn.initializer.Normal 
后面同理

File: paddlevideo/modeling/backbones/ffa_net.py [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+                                        math.sqrt(channel // 8 * (1**2))))),
+            nn.Sigmoid())
+
+    def forward(self, x):
+        y = self.pa(x)
+        return x * y
+
+
+@BACKBONES.register()
+class CALayer(nn.Layer):  #通道注意力块
```
Comment (huangjun12) [inline paddlevideo/modeling/backbones/ffa_net.py:78]: 中文注释改成英文

File: paddlevideo/modeling/builder.py [hunk 11]
```diff
@@ -104,6 +105,11 @@ def build_segment(cfg):
     return build(cfg, SEGMENT, key='framework')
 
 
+def build_generator(cfg):
+    """Build recognizer."""
```
Comment (huangjun12) [inline paddlevideo/modeling/builder.py:109]: 注释 generator

**Review by huangjun12 — COMMENTED** [review]
中文注释改成英文，删除无效注释，各个文件都check下吧

**Comment by huangjun12** [comment]
还有加一下tipc，
参考https://github.com/PaddlePaddle/models/blob/release%2F2.2/tutorials/tipc/train_infer_python/README.md
加到这里 https://github.com/PaddlePaddle/PaddleVideo/tree/develop/test_tipc

**Pull request closed without merging**